drivers/flash/flash_stm32_fmc_mt29f4g08.c

```diff
@@ -0,0 +1,336 @@
+		.spare_area_size = config->spare_area_size,
+		.block_size = config->block_size,
+		.plane_size = config->plane_size,
+		.flash_size = config->flash_size,
+		.setup_time = config->setup_time,
+		.wait_setup_time = config->wait_setup_time,
+		.hold_setup_time = config->hold_setup_time,
+		.hiz_setup_time = config->hiz_setup_time,
+	};
+
+	/* Initialise NAND bank */
+	ret = flash_stm32_fmc_nand_init_bank(controller, &init);
+	if (ret != 0) {
+		LOG_ERR("NAND bank initialisation failed with error %d", ret);
+		return -EIO;
+	}
+
+	/* Reset NAND flash */
+	ret = flash_stm32_fmc_nand_reset(controller);
+	if (ret != 0) {
+		LOG_ERR("NAND flash reset failed with error %d", ret);
+		return -EIO;
+	}
+
+#ifdef CONFIG_FLASH_MT29F4G08_ECC
+	/* Enable on-die ECC feature */
+	struct nand_flash_feature ecc_feature = {
+		.feature_addr = ECC_FEATURE_ADDR,
+		.feature_data = ECC_FEATURE_DATA,
+	};
+	ret = flash_stm32_fmc_nand_set_feature(controller, &ecc_feature);
+	if (ret != 0) {
+		LOG_ERR("Enabling on-die ECC failed with error %d", ret);
+		return -EIO;
+	}
+#endif /* CONFIG_FLASH_MT29F4G08_ECC */
+
+	LOG_INF("MT29F4G08 flash initialised with FMC controller %s", controller->name);
+
+	return 0;
+}
+
+static DEVICE_API(flash, flash_stm32_fmc_mt29f4g08_api) = {
+	.read = flash_mt29f4g08_read,
+	.write = flash_mt29f4g08_write,
+	.erase = flash_mt29f4g08_erase,
+	.get_parameters = flash_mt29f4g08_get_parameters,
+	.get_size = flash_mt29f4g08_get_size,
+#ifdef CONFIG_FLASH_PAGE_LAYOUT
+	.page_layout = flash_mt29f4g08_page_layout,
+#endif
+#ifdef CONFIG_FLASH_EX_OP_ENABLED
+	.ex_op = flash_mt29f4g08_ex_op,
+#endif /* CONFIG_FLASH_EX_OP_ENABLED */
+};
+
+/* A page in this context corresponds to the smallest erasable area which is a block */
+#define LAYOUT_PAGES_PROP(n)                                                                       \
+	IF_ENABLED(CONFIG_FLASH_PAGE_LAYOUT,                                                       \
+		(.layout = {                                                                       \
+			.pages_count = DT_INST_PROP(n, flash_size) /                               \
+				       DT_INST_PROP(n, block_size),                                \
+			.pages_size = DT_INST_PROP(n, block_size),                                 \
+		}))
+
+#define FLASH_STM32_FMC_MT29F4G08_INIT(n)                                                          \
+	static const struct flash_mt29f4g08_config flash_stm32_fmc_mt29f4g08_config_##n = {        \
+		.controller = DEVICE_DT_GET(DT_INST_PARENT(n)),                                    \
+		.parameters =                                                                      \
+			{                                                                          \
+				.write_block_size = DT_INST_PROP(n, page_size),                    \
+				.erase_value = 0xff,                                               \
+			},                                                                         \
```

Comment:
```suggestion
		.parameters = {                                                                        \
			.write_block_size = DT_INST_PROP(n, page_size),                            \
			.erase_value = 0xff,                                                       \
		},                                                                                  \

Reply:
Formatter puts it that way. Following change would work:
```
		.parameters = {.write_block_size = DT_INST_PROP(n, page_size),                     \
				       .erase_value = 0xff},                                               \
```